### PR TITLE
Add front end testing

### PR DESCRIPTION
**Pull request by SteveBarnett**
Add a whole bunch of FED testing stuff using [DalekJS](http://dalekjs.com/).

[NightwatchJS](http://nightwatchjs.org/) turned out to be quite fiddly with the set up of Selenium and stuff.

[CasperJS](http://casperjs.org/) didn't look like it was going to play nicely with NodeJS (and was giving me PhantomJS-related trouble on OSX).

@avermeulen, @Onwa2014, @Oyamasiphula : for review, please!

### CODE REVIEW

File: front_end_testing.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+---
+layout: default
+title: Front-end testing
+---
+
+# Front-end testing
```
Comment (avermeulen) [inline front_end_testing.md]: @SteveBarnett should this be a separate sub section?

File: 7_Search_and_filter.md [hunk 2]
```diff
@@ -21,6 +22,26 @@ Add support for:
 * finding sales by product or category name;
 * finding purchases by product or category name.
 
+## Front-end testing
+
+Add some more DalekJS tests. You'll need some new methods like `type` and `submit`.
```
Comment (avermeulen) [inline 7_Search_and_filter.md]: @SteveBarnett It's not clear where should these new methods be added.

File: front_end_testing.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Sometimes we want to do UI (User Interface) testing. One thing we can use for this is [DalekJS](http://dalekjs.com/). We'll use it to test that the right pages are there, and that they work as expected.
+
+## Set up DalekJS
+
+DalekJS is available as an npm package, which makes installation convenient for us. Follow [the installation instructions](http://dalekjs.com/pages/getStarted.html#install).
+
+Install the dalek-cli globally ([like we did for the Mocha cli](http://node.projectcodex.co/node-tutorial/index.html)).
+
+```
+npm install dalek-cli -g
```
Comment (avermeulen) [inline front_end_testing.md]: @SteveBarnett this global install will need  to be done a super user on ubuntu `sudo` command so will need a `sudo`.

File: 7_Search_and_filter.md (hunk 2 above)
Comment (SteveBarnett) [inline 7_Search_and_filter.md]: @avermeulen Cool, thanks. I've updated it here: https://github.com/codex-academy/Nelisa/pull/16/commits/86bb6a35ee005a641c100314c43e3679e31c00f1

File: front_end_testing.md (hunk 3 above)
Comment (SteveBarnett) [inline front_end_testing.md]: @avermeulen Good catch, thanks! Fixed in https://github.com/codex-academy/Nelisa/pull/16/commits/9b6c3d473df970b54e22eb75ae9690a4775ee232 .

File: 7_Search_and_filter.md (hunk 2 above)
Comment (SteveBarnett) [inline 7_Search_and_filter.md]: And here: https://github.com/codex-academy/Nelisa/pull/16/commits/c84e7833d87521671f5dffc65a9f8ea9bed6cb4c

**Comment by SteveBarnett** [comment]
@avermeulen can you review these updates please?

**Comment by SteveBarnett** [comment]
@avermeulen Are you still looking at this? I've pulled in the latest `gh-pages` so it can be merged.

**Comment by avermeulen** [comment]
@SteveBarnett I still have a sticky against this to get Dalek running. 

@Oyamasiphula can you please have a look to see if you can get DalekJS running on Ubuntu? As per Steve's instructions.

**Comment by SteveBarnett** [comment]
Closing, but leaving the branch.

**Pull request closed without merging**